src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -187,43 +248,49 @@ private static IQueryMatch ToCoraxQuery(Parameters builderParameters, QueryExpre
 
                             TryMergeTwoNodesForAnd(indexSearcher, builderParameters.AllEntries, ref left, ref right, out var merged, true);
 
                             return indexSearcher.AndNot(builderParameters.AllEntries.Replay(), indexSearcher.Or(left, right), token: builderParameters.Token);
 
                         case (NegatedExpression ne1, _):
-                            left = ToCoraxQuery(builderParameters, @where.Right, exact);
-                            right = ToCoraxQuery(builderParameters, ne1.Expression, exact);
+                            left = ToCoraxQuery(builderParameters, @where.Right, ref builderParameters.StreamingDisabled, exact);
+                            right = ToCoraxQuery(builderParameters, ne1.Expression, ref builderParameters.StreamingDisabled, exact);
 
                             TryMergeTwoNodesForAnd(indexSearcher, builderParameters.AllEntries, ref left, ref right, out merged, true);
 
                             return indexSearcher.AndNot(right, left, token: builderParameters.Token);
 
                         case (_, NegatedExpression ne1):
-                            left = ToCoraxQuery(builderParameters, @where.Left, exact);
-                            right = ToCoraxQuery(builderParameters, ne1.Expression, exact);
+                            left = ToCoraxQuery(builderParameters, @where.Left, ref builderParameters.StreamingDisabled, exact);
+                            right = ToCoraxQuery(builderParameters, ne1.Expression, ref builderParameters.StreamingDisabled, exact);
 
                             TryMergeTwoNodesForAnd(indexSearcher, builderParameters.AllEntries, ref left, ref right, out merged, true);
                             return indexSearcher.AndNot(left, right, token: builderParameters.Token);
 
                         default:
-                            left = ToCoraxQuery(builderParameters, @where.Left, exact);
-                            right = ToCoraxQuery(builderParameters, @where.Right, exact);
-
+                            left = ToCoraxQuery(builderParameters, @where.Left, ref streamingOptimization, exact);
+                            right = ToCoraxQuery(builderParameters, @where.Right, ref builderParameters.StreamingDisabled, exact);
 
+                            if (left is CoraxBooleanItem cbi && streamingOptimization.TrySetAsStreamingField(cbi))
+                                left = cbi.Materialize();
+                            
                             if (TryMergeTwoNodesForAnd(indexSearcher, builderParameters.AllEntries, ref left, ref right, out merged))
                                 return merged;
 
+                            // When we do not optimize this node for streaming operation we can swap nodes in BinaryMatch to be more efficient based on term counts.
+                            if (streamingOptimization.OptimizationIsPossible == false && left.Count < right.Count && left.Confidence >= QueryCountConfidence.Normal)
```

Comment:
Add a bit more context on what this optimization is about. We pull that one from the IndexSearcher, so we should make it clear which ticket was the optimization history based on (if any).

Reply:
Yes, I can't understand the intent here at a glance.

Reply:
Okay. Added also a previous comment.
This originally was part of `IndexSearcher` but when we do streaming we've to ensure our streaming node is left-most, so we cannot "swap" branches.